nncf/common/graph/patterns.py

```diff
@@ -107,9 +107,9 @@ def __add__(self, other: 'GraphPattern') -> 'GraphPattern':
                 # add merge all possible connections
                 # A: (a) (b)
                 # B: (c) (d)
-                #              (a)  (a_copy)  (b)    (b_copy)
+                #              (a_copy)  (a_copy)  (b_copy)    (b_copy)
                 # A + B ---->   |       |      |        |
-                #              (c)     (d)  (c_copy) (d_copy)
+                #              (c_copy)     (d_copy)  (c_copy) (d_copy)
```

Comment:
Is it the correct change?

Reply:
The previous comment was incorrect, I fixed it.
In `__add__` magic method we create a new graph and copy every weakly component for both initial graphs (self, other), connect them and put into the final graph.
In this logic all weakly connected components are copies.

Reply:
Seems like we need to align `|` as well.

Reply:
Done